### PR TITLE
Jormungandr: documentation of /places

### PR DESCRIPTION
**Pull request by antoine-de**
add the swagger documentation to /places

I updated flask_restful to the lastest version to be able to avoid method decorators on `OPTIONS`

the descriptions on all the output needs to be done

the swagger output needs to be better checked :wink:

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/parsers.py [hunk 1]
```diff
@@ -76,20 +78,24 @@ def unsigned_integer(value):
...
+        if not (-90.0 <= lat <= 90.0):
+            raise ValueError("lat should be between -90 and 90")
+        lon = float(lon)
+        if not (180.0 >= lon >= -180.0):
+            raise ValueError("lon should be between -180 and 180")
+
+        return coord
+
+    def description(self):
+        return ParameterDescription(type=str, metadata={'pattern': '.*;.*'})
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/parsers.py:101]: genuine question (don't know where the cursor is): don't we wanna precise more that it's float?

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 2]
```diff
@@ -61,38 +66,38 @@
...
+class geojson_argument(object):
+    def __call__(self, value):
+        decoded = json.loads(value)
+        if not decoded:
+            raise ValueError('invalid shape')
+
+        return parser_args_type.geojson_argument(decoded)
+
+    def description(self):
+        return ParameterDescription(type=str)  # TODO a better description of the geojson
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py:78]: maybe just add a link in description to https://geojson.org/geojson-spec.html (or the RFC if readable enough) ?

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 3]
```diff
@@ -61,38 +66,38 @@
...
         self.parsers["get"].add_argument("count", type=default_count_arg_type, default=10,
                                          description="The maximum number of places returned")
         self.parsers["get"].add_argument("search_type", type=int, default=0,
                                          description="Type of search: firstletter or type error")
         self.parsers["get"].add_argument("admin_uri[]", type=unicode,
                                          action="append",
-                                         description="If filled, will\
-                                         restrained the search within the\
-                                         given admin uris")
+                                         description="If filled, will restrained the search within the "
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py]: `restrain` (no "-ed")

**Review by pbougue — COMMENTED** [review]
seems ok, but not sure I have a sharp eye on that

File: source/jormungandr/jormungandr/interfaces/parsers.py (hunk 1 above)
Comment (antoine-de) [reply source/jormungandr/jormungandr/interfaces/parsers.py:101]: I'll let you write the regex for a float :grin: 

more seriously I think we should not overengineer this sort of stuff and wait for the SDK input to fit their needs

File: source/jormungandr/jormungandr/interfaces/v1/Places.py (hunk 2 above)
Comment (antoine-de) [reply source/jormungandr/jormungandr/interfaces/v1/Places.py:78]: hum I don't think we can do this, I don't see how the SDK will be able to handle this parameter (but maybe we don't need it and we can do a simple thing as giving a link to the documentation as you said)

**Pull request merged**